src/components/StartHostedActivitySessionFlow/StartHostedActivitySessionFlow.tsx

```diff
@@ -28,10 +29,12 @@ export const StartHostedActivitySessionFlow: FC<
   }
 
   useEffect(() => {
-    if (!isNil(data?.sessionId)) {
-      router.replace(`../?sessionId=${data?.sessionId}`)
+    if (!isNil(data?.sessionId) && !isNil(data?.language)) {
+      const { sessionId, language } = data
+      const language_prefix = mapLanguageToLocalePrefix({ language })
```

Comment:
See the previous comment, it seems we can let the back-end (API) take care of returning the correct URL including the language instead of handling this in the front-end.